pyuvdata/tests/test_uvdata.py

```diff
@@ -2936,3 +2936,45 @@ def test_redundancy_finder_when_nblts_not_nbls_times_ntimes():
     redundant_groups, centers, lengths, conj_inds = uv.get_baseline_redundancies(tol=tol)
     redundant_groups.sort(key=len)
     nt.assert_equal(groups, redundant_groups)
+
+
+def test_overlapping_data_add():
+    # read in test data
+    uv = UVData()
+    testfile = os.path.join(DATA_PATH, 'day2_TDEM0003_10s_norx_1src_1spw.uvfits')
+    uvtest.checkWarnings(uv.read_uvfits, [testfile], message='Telescope EVLA is not')
+
+    # slice into four objects
+    blts1 = np.arange(500)
+    blts2 = np.arange(500, 1360)
+    uv1 = uv.select(polarizations=[-1, -2], blt_inds=blts1, inplace=False)
+    uv2 = uv.select(polarizations=[-3, -4], blt_inds=blts1, inplace=False)
+    uv3 = uv.select(polarizations=[-1, -2], blt_inds=blts2, inplace=False)
+    uv4 = uv.select(polarizations=[-3, -4], blt_inds=blts2, inplace=False)
+
+    # combine and check for equality
+    uvfull = uv1 + uv2
+    uvfull += uv3
+    uvfull += uv4
+    extra_history = ("  Downselected to specific baseline-times, polarizations using pyuvdata. "
+                     "Combined data along polarization axis using pyuvdata. Combined data along "
+                     "baseline-time axis using pyuvdata. Overwrote invalid data using pyuvdata.")
+    nt.assert_equal(uvfull.history, uv.history + extra_history)
+    uvfull.history = uv.history  # make histories match
+    nt.assert_equal(uv, uvfull)
+
+    # check combination not-in-place
+    uvfull = uv1 + uv2
+    uvfull += uv3
+    uvfull = uvfull + uv4
+    uvfull.history = uv.history  # make histories match
+    nt.assert_equal(uv, uvfull)
```

Comment:
I like these tests, but I also want to be sure that if the are written out as separate files and all 4 filenames are passed to the generic `read` method that there are no errors and it works as expected. I think that's the most common way this code will actually be tripped.

It's a little interesting that in that case I think there will still be something in the history about overwriting invalid data, even though there was no invalid data in any of the individual files. I'm not sure whether I'm worried about that or not.

Reply:
Good thought! I've added that test.

Yes, it does still add the "invalid data" string to the history. I suppose one option could be to inspect the stack and see if the parent function is a `read*` method, and not add the string in that case. Otherwise I'm not sure how we can tell whether it's worth adding or not.

Reply:
Yeah, I think we leave it. It just struck me. Thanks for adding the test.